### PR TITLE
fix(proxy-template): Correct bad merging of duration

### PR DESCRIPTION
**Pull request by lahabana**
### Summary

We used to merge 10s and 0.5s into 10.5s this is because
proto.Duration is made of 2 fields (seconds and nanos) and the default
merge method would just add these 2 fields separately.
### Full changelog

This patch copies the content of proto.Merge but enables us
to override the merge policy for specific types.

### Testing

- [x] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

I don't really want to backport this as it's really a change in behaviour...

### CODE REVIEW

File: pkg/util/proto/proto.go [hunk 1]
```diff
@@ -96,3 +99,112 @@ func MustToStruct(message proto.Message) *structpb.Struct {
...
+	})
+}
+
+func MergeForKuma(dst, src proto.Message) {
+	duration := &durationpb.Duration{}
+	Merge(dst, src, MergeFunctionOptionFn(duration.ProtoReflect().Descriptor().FullName(), ReplaceMergeFn))
+}
+
+// Merge Code of proto.Merge with modifications to support custom types
+func Merge(dst, src proto.Message, opts ...OptionFn) {
```
Comment (jpeach) [inline pkg/util/proto/proto.go:134]: So there's a public api "Merge" and also "MergeForKuma". Can we make "Merge" the real api and leave the rest as implementation details?

File: pkg/util/proto/proto.go [hunk 2]
```diff
@@ -96,3 +99,112 @@ func MustToStruct(message proto.Message) *structpb.Struct {
...
+		return true
+	})
+}
+
+func MergeForKuma(dst, src proto.Message) {
+	duration := &durationpb.Duration{}
+	Merge(dst, src, MergeFunctionOptionFn(duration.ProtoReflect().Descriptor().FullName(), ReplaceMergeFn))
+}
+
+// Merge Code of proto.Merge with modifications to support custom types
```
Comment (jpeach) [inline pkg/util/proto/proto.go:133]: If we are copying code in from somewhere else, we probably should preserve copyright and licenses?

**Comment by lahabana** [comment]
I've opened an issue upstream as well: https://github.com/golang/protobuf/issues/1359

**Comment by lahabana** [comment]
@jakubdyszkiewicz @jpeach @parkanzky I feel like this shouldn't be backported as it's in practice a breaking change (even if the previous behaviour is far from what's expected).

WDYT?

**Comment by lahabana** [comment]
Discussed offline and decided we should backport

**Pull request merged**